src/validate.js

```diff
@@ -1,10 +1,33 @@
 import Ajv from 'ajv';
 import ajvKeywords from 'ajv-keywords';
 
-import absolutePathKeyword from './keywords/absolutePath';
+import addAbsolutePathKeyword from './keywords/absolutePath';
 
 import ValidationError from './ValidationError';
 
+/** @typedef {import("json-schema").JSONSchema4} JSONSchema4 */
+/** @typedef {import("json-schema").JSONSchema6} JSONSchema6 */
+/** @typedef {import("json-schema").JSONSchema7} JSONSchema7 */
+/** @typedef {import("ajv").ErrorObject} ErrorObject */
+
+/** @typedef {(JSONSchema4 | JSONSchema6 | JSONSchema7) & { patternRequired?: Array<string>; formatMinimum?: string; formatMaximum?: string; formatExclusiveMinimum?: boolean; formatExclusiveMaximum?: boolean; instanceof?: string | Array<string>; if?: Schema | boolean; then?: Schema | boolean; else?: Schema | boolean}} Schema */
```

Comment:
Unfortunately, it will not work as expected. For example for arrays https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/json-schema/index.d.ts#L104
TS will warn on custom keywords =(
Don't know how to provide types here better, maybe create custom JSON definitions with supporting template. Like:

in json schema definition
```
type JSONSchema4<T as {[k: string]: string | number | boolean | string[] | number[] | boolean[]}> = {
       items?: Array<JSONSchema4<T>>
} & T

then
```
type AdditionalKeywords = {formatMinimum?: string ...};
type Schema = JSONSchema4<AdditionalKeywords>
``

Reply:
What about this solution https://github.com/epoberezkin/ajv-keywords/issues/53?

Reply:
I afraid here no good solutions, maybe we can use `/** typeof /** @type {{ patternRequired: string }} **/ (schema).patternRequired !== 'undefined' **/`, what do you think, also we can improve this in future